runtime/lua/vim/treesitter.lua

```diff
@@ -85,17 +86,20 @@ function M.get_parser(bufnr, lang, opts)
   if bufnr == nil or bufnr == 0 then
     bufnr = a.nvim_get_current_buf()
   end
-  if lang == nil then
+  if not lang then
     lang = a.nvim_buf_get_option(bufnr, "filetype")
   end
 
-  if parsers[bufnr] == nil then
-    parsers[bufnr] = M._create_parser(bufnr, lang, opts)
+  local parser = parsers[bufnr] and parsers[bufnr][lang]
+  if not parser then
+    parser = M._create_parser(bufnr, lang, opts)
+    parsers[bufnr] = parsers[bufnr] or {}
```

Comment:
I don't really like that way of doing it?

```suggestion
    if not parsers[bufnr] then
      parsers[bufnr] = {}
    end
```